pkg/graph/dag/dag.go

```diff
@@ -23,62 +24,75 @@ import (
 )
 
 // Vertex represents a node/vertex in a directed acyclic graph.
-type Vertex struct {
+type Vertex[T cmp.Ordered] struct {
 	// ID is a unique identifier for the node
-	ID string
+	ID T
 	// Order records the original order, and is used to preserve the original user-provided ordering as far as posible.
 	Order int
 	// DependsOn stores the IDs of the nodes that this node depends on.
 	// If we depend on another vertex, we must appear after that vertex in the topological sort.
-	DependsOn map[string]struct{}
+	DependsOn map[T]struct{}
 }
 
-func (v Vertex) String() string {
-	dependsOn := strings.Join(maps.Keys(v.DependsOn), ",")
-	return fmt.Sprintf("Vertex[ID: %s, Order: %d, DependsOn: %s]", v.ID, v.Order, dependsOn)
+func (v Vertex[T]) String() string {
+	var builder strings.Builder
+	for i, s := range maps.Keys(v.DependsOn) {
+		builder.WriteString(fmt.Sprintf("%v", s))
+		if i < len(v.DependsOn)-1 {
+			builder.WriteString(",")
+		}
+	}
```

Comment:
Not sure how different this is from just using `strings.Join(maps.Keys(v.DependsOn), ",")`

Reply:
The only thing i could think of is performance, but looks like `strings.Join` uses `strings.Builder` under the hood https://cs.opensource.google/go/go/+/refs/tags/go1.24.1:src/strings/strings.go;l=481-513

Reply:
I was concerned about that too. I didnt want to compare the entire slice to strings so Thats why I ended up with the builder. I adopted the Growing of the Builder capacity from strings.Join now to avoid additional allocs like in the stdlib function